src/cargo/ops/cargo_rustc/compilation.rs

```diff
@@ -109,12 +109,17 @@ impl<'cfg> Compilation<'cfg> {
             }
         }
 
+	 let metadata = pkg.manifest().metadata().clone();
```

Comment:
If it doesn't return a reference then you shouldn't need the `clone()` as it's not a reference, but if it does a reference you should be able to use the fields through the reference (without cloning).

The parts below can probably be:

```rust
metadata.description.as_ref().unwrap_or(&String::new())
```